Source/WebCore/dom/Node.cpp

```diff
@@ -808,7 +797,17 @@ Node::Editability Node::computeEditability(UserSelectAllTreatment treatment, Sho
         document->updateStyleIfNeeded();
     }
 
-    return computeEditabilityFromComputedStyle(*this, treatment, pageIsEditable);
+    if (!style)
+        style = isDocumentNode() ? renderStyle() : const_cast<Node*>(this)->computedStyle();
```

Comment:
Can these bits be moved to the method that doesn't take `RenderStyle*` ? Ideally this method would take `RenderStyle&` though.

Reply:
I didn't want to do that in order to avoid the computedStyle() in all of the cases where we early-return. It's my understanding that this method is fairly hot in some benchmarks, and that seemed like a bad idea.